include/zephyr/device.h

```diff
@@ -304,6 +304,20 @@ typedef int16_t device_handle_t;
  */
 #define DEVICE_GET(dev_id) (&DEVICE_NAME_GET(dev_id))
 
+/**
+ * @brief Declare a devicetree device object.
+ *
+ * This macro will assert at build time if the given @p node_id is not found or
+ * not enabled in Devicetree.
+ *
+ * @param node_id Devicetree node identifier.
+ */
+#define DEVICE_DT_DECLARE(node_id)                                             \
+		BUILD_ASSERT(DT_NODE_HAS_STATUS(node_id, okay),                \
+			     "Device " STRINGIFY(node_id)                      \
+			     " not found or not enabled in Devicetree");       \
```

Comment:
You can clean up the error message as follows:

1. if device node exists, print its full path and say it isn't enabled
2. otherwise, say the node identifier is invalid

Reply:
done